📖 Update Try-it page based on the v1beta1 API

ren-yifei opened the pull request:
Update the Try-it.md to include the changes from release v1beta1.
The session 2.5 Running custom CAPM3 is not finished, since the
make run command is deprecated.

mboukhalfa commented on try-it.md:
```diff
@@ -286,9 +288,13 @@ status:
 This section describes how to trigger provisioning of a cluster and hosts via
 `Machine` objects as part of the Cluster API integration. This uses Cluster API
-[v1alpha4](https://github.com/kubernetes-sigs/cluster-api/tree/v0.3.0) and
+<<<<<<< HEAD
```
should be fixed

Xenwar commented on try-it.md:
```diff
@@ -534,73 +516,77 @@ Example:
++--------------------------------------+---------------+--------------------------------------+-------------+--------------------+-------------+
+| UUID                                 | Name          | Instance UUID                        | Power State | Provisioning State | Maintenance |
++--------------------------------------+---------------+--------------------------------------+-------------+--------------------+-------------+
+| b423ee9c-66d8-48dd-bd6f-656b93140504 | metal3~node-1 | 7f51f14b-7701-436a-85ba-7dbc7315b3cb | power off   | available          | False       |
+| 882533c5-2f14-49f6-aa44-517e1e404fd8 | metal3~node-0 | 14fbcd25-4d09-4aca-9628-a789ba3e175c | power off   | available          | False       |
```
```suggestion
| b423ee9c-66d8-48dd-bd6f-656b93140504 | node-1 | 7f51f14b-7701-436a-85ba-7dbc7315b3cb | power off   | available          | False       |
| 882533c5-2f14-49f6-aa44-517e1e404fd8 | node-0 | 14fbcd25-4d09-4aca-9628-a789ba3e175c | power off   | available          | False       |
```

Xenwar replied:
not sure if there is a change to the naming of bmhs

lentzi90 commented on try-it.md:
````diff
@@ -480,36 +474,24 @@ First, you must scale down the deployment of the `baremetal-operator` running
-cd ~/go/src/github.com/metal3-io/cluster-api-provider-metal3
-make run
-```
+controller for Metal3. You can try changes to
+`cluster-api-provider-metal3` and the changes will be automaticly applied.
````
This sounds strange. Is it when using Tilt? In that case I think it should be mentioned. E.g. "... changes will be automatically applied when using Tilt."

lentzi90 commented on try-it.md:
```diff
@@ -32,10 +32,12 @@ permalink: /try-it.html
 - Run as a user with passwordless sudo access
-- Minimum resource requirements for the host machine: 4C CPUs, 16 GB RAM memory.
+- Minimum resource requirements for the host machine: 4C CPUs, 16 GB RAM memory
+- The host machine should have a floating IP
```
I think this is not necessary but please correct me if I'm wrong. It should be possible to run all of this locally without any floating IP.

lentzi90 commented on try-it.md:
````diff
@@ -480,36 +474,24 @@ First, you must scale down the deployment of the `baremetal-operator` running
 
 ```sh
-cd ~/go/src/github.com/metal3-io/baremetal-operator
-make run
+$ cd ~/go/src/github.com/metal3-io/baremetal-operator
+$ make run
````
No big deal, but when including the `$` prompt, it would be more correct to use `console` or `shell` instead of `sh`. I noticed that this guide is using `sh` all the time and I think it would be nice to change it to `console`. It makes the syntax highlighting correct.

See https://stackoverflow.com/a/49004070 .

lentzi90 reviewed (commented):
Good to see this updated!

lentzi90 reviewed (approved):
Great job!

Edit: remember to squash the commits before merging!

ren-yifei commented:
/assign @hardys

fmuyassarov commented on try-it.md:
```diff
@@ -286,14 +287,14 @@ status:
+[v1beta1](https://github.com/kubernetes-sigs/cluster-api/tree/v1.0.2) and
 assumes that metal3-dev-env is deployed with the environment variable
-**CAPM3_VERSION** set to **v1alpha4**. This is the default behavior. The v1alpha4 deployment can be done with
+**CAPM3_VERSION** set to **v1beta1**. This is the default behavior. The v1beta1 deployment can be done with
 Ubuntu 18.04 or Centos 8 target host images. Please make sure to meet [resource requirements](#11-prerequisites) for successful deployment:
```
```suggestion
**CAPM3_VERSION** set to **v1beta1**. This is the default behavior. The v1beta1 deployment can be done with
Ubuntu 20.04 or Centos 8 target host images. Please make sure to meet [resource requirements](#11-prerequisites) for successful deployment:
```

fmuyassarov commented on try-it.md:
````diff
@@ -479,37 +472,25 @@ image built from the [metal3-io/baremetal-operator](https://github.com/metal3-io
-make run
-```
+controller for Metal3. You can try changes to
+`cluster-api-provider-metal3` and the changes will be automaticly applied when using Tilt.
````
we haven't talked about Tilt until this point and not sure if the mention of it will not confuse people.
I think either we need to explain how to work with Tilt/give a reference to a document where we have explained or
keep this part as it is

fmuyassarov commented:
Also I would keep code blocks as shell format instead of console. They look a bit different in the preview website: https://deploy-preview-224--metal3.netlify.app/try-it.html

lentzi90 commented:
> Also I would keep code blocks as shell format instead of console. They look a bit different in the preview website: https://deploy-preview-224--metal3.netlify.app/try-it.html

But shell gives correct syntax highlighting when the prompt is included. Then it knows what is output (no syntax highlighting) and what is the command (with syntax highlighting).
There is one part of the page where `console` is used wrong though: https://github.com/metal3-io/metal3-io.github.io/blob/f1fcf31f378814901ae622ec829032e31c0d3f23/try-it.md?plain=1#L434-L451
Here the comments (starting with `#`) are treated as root prompt. (`$` is normal user prompt, `#` is root). If we want these highlighted as comments they should have the prompt in front (`$ # Comment here`) or we switch that block to shell.

I vote for putting the prompt before the comment. Then it is obvious what part is output and what is input. :slightly_smiling_face:

ren-yifei commented:
I updated the code blocks part using console, and improved the comments lines in the block, please check whether that looks better or not.
Also I added a brief introduction part for Tilt, please check that all the information is correct or not.

fmuyassarov reviewed (commented):
/lgtm
/cc @hroyrh

fmuyassarov commented:
/hold
@ren-yifei can you please squash commits before we merge it?

furkatgofurov7 commented on try-it.md:
```diff
@@ -479,37 +472,37 @@ image built from the [metal3-io/baremetal-operator](https://github.com/metal3-io
 
-Then you can run the custom Machine controller manager out of your local git tree.
+[Tilt](https://tilt.dev) setup can deploy CAPM3 in a local kind cluster. Since
+Tilt is applied in the metal3-dev-env deployment, you can try changes to
+`cluster-api-provider-metal3` and Tilt will deploy the changes automaticly.
```
```suggestion
Tilt is applied in the metal3-dev-env deployment, you can make changes inside the
`cluster-api-provider-metal3` folder and Tilt will deploy the changes automatically.
```

furkatgofurov7 commented on try-it.md:
```diff
@@ -32,10 +32,11 @@ permalink: /try-it.html
 ### 1.1. Prerequisites
 
-- System with CentOS 8 or Ubuntu 18.04
+- System with CentOS 8 or Ubuntu 20.04
```
```suggestion
- System with Centos 8 Stream or Ubuntu 20.04
```

furkatgofurov7 commented on try-it.md:
```diff
@@ -32,10 +32,11 @@ permalink: /try-it.html
 - Run as a user with passwordless sudo access
-- Minimum resource requirements for the host machine: 4C CPUs, 16 GB RAM memory.
+- Minimum resource requirements for the host machine: 4C CPUs, 16 GB RAM memory
+- Required dependencies: make, kubelet, kubeadm, kubectl, go (version 1.17.5)
```
I would suggest not to include these as dependencies since most of them will be installed by m3-dev-env scripts

furkatgofurov7 commented on try-it.md:
```diff
@@ -286,14 +287,14 @@ status:
-**CAPM3_VERSION** set to **v1alpha4**. This is the default behavior. The v1alpha4 deployment can be done with
-Ubuntu 18.04 or Centos 8 target host images. Please make sure to meet [resource requirements](#11-prerequisites) for successful deployment:
+**CAPM3_VERSION** set to **v1beta1**. This is the default behavior. The v1beta1 deployment can be done with
+Ubuntu 20.04 or Centos 8 target host images. Please make sure to meet [resource requirements](#11-prerequisites) for successful deployment:
```
```suggestion
Ubuntu 20.04 or Centos 8 Stream target host images. Please make sure to meet [resource requirements](#11-prerequisites) for successful deployment:
```

furkatgofurov7 commented on try-it.md:
```diff
@@ -479,37 +472,36 @@ image built from the [metal3-io/baremetal-operator](https://github.com/metal3-io
+Tilt is applied in the metal3-dev-env deployment, you can make changes inside
+the `cluster-api-provider-metal3` folder and Tilt will deploy the changes
+automatically.
+If you deployed CAMP3 separately and want to make changes in CAPM3, then
```
```suggestion
If you deployed CAPM3 separately and want to make changes to it, then
```

furkatgofurov7 reviewed (changes requested):
Overall lgtm, small nit that needs a fix:

furkatgofurov7 commented:
/retitle 📖 Update Try-it page based on the v1beta1 API

furkatgofurov7 reviewed (commented):
LGTM

fmuyassarov commented:
/lgtm

ren-yifei commented:
/assign @hroyrh

furkatgofurov7 commented:
/hold cancel
/assign @knowncitizen @hroyrh @hardys
